front-end/src/Components/Fucntions.js

```diff
@@ -0,0 +1,116 @@
+export const onChangeHandler = (event, data, setData) => {
+    const temp = {};
+    temp[event.target.name] = event.target.value;
+
+    if (event.target.name === "Gender")
+        temp[event.target.name] = event.target.id;
+
+    if (data)
+        setData({ ...data, ...temp });
+    else
+        setData(temp);
+}
+
+export const onClickHandler = async (route, method, navigate, data) => {
+    console.log("in the on click handler for the "+method);
+    const response = await fetch(`/${route}`, {
+        method: method,
```

Comment:
what is this method is this a variable?
if this is a varibale then you don't need to write it in a variable unless it is not using in more than one place you can simply write POST or GET etc

Reply:
okay I will see to that